src/parser.cpp

```diff
@@ -4,11 +4,76 @@
 #include "unicode.cpp"
 #include "scheme.cpp"
 
-#include <cctype>
 #include <algorithm>
+#include <cctype>
+#include <string_view>
```

Comment:
consider `#include <cstring>`.

Reply:
For education purposes: What is the reason behind this?